ome/ome-model
# OME element implements OMEModel

This is part 1 of the proposed refactoring in https://docs.google.com/document/d/1bzczgG2QyujCsCvR2kSAGl5hj-NJuIikgFAC3fDkqvs/edit# and https://trello.com/c/hnpxmvhe/52-extended-annotation-support-pt-i

- OME implements OMEModel
- OMEModelImpl (Java) and detail::OMEModel (C++) are no longer used and are deprecated; OME implements OMEModel directly, and OMEModelObject indirectly
- MetadataRoot replaced by MetadataModel
- OMEXMLMetadataRoot replaced by OMEXMLMetadataModel

Testing:

- Check generated source changes
- Check builds remain green
- Check bioformats build with and without https://github.com/openmicroscopy/bioformats/pull/2712
- Check ome-files-cpp build with and without https://github.com/rleigh-codelibre/ome-files-cpp/tree/ome-model-base (this is PR https://github.com/ome/ome-files-cpp/pull/40)

The latter two points check that there is no API break when building without the branch, and that the new API additions work when building with the branch,
